### DISCUSSION TITLE
First Time Contributer - Where Should I Work on My Contribution?

### DISCUSSION THREAD

**Discussion by GalMichaeli** [networkx/networkx#7744] [Q&A]
Hi,
I'm working on implementing several Connected Dominating Set (CDS) algorithms. I've noticed that there are two locations in the main repository for Dominating Set algorithms, namely
(a) networkx/networkx/algorithms/dominating.py
and,
(b) networkx/networkx/algorithms/approximation/dominating_set.py

The algorithms I'm working on are considered approximation algorithms and are some of the earlier ones introduced in the literature, specifically the ones specified [here](https://typeset.io/pdf/approximation-algorithms-for-connected-dominating-sets-2slkc2r90s.pdf).

I'd like to know where it would be best to work on my contribution, file (a) or (b).
Thanks.

**Comment by dschult** [comment]
Algorithms for approximate solutions should be added in (b).
Thanks!
